Add Progressbar to Milestone Page

JakobDev opened the pull request:
This is adds the progress bar, which is already on the Milestone List, also to the Page of a Single Milestone.

![grafik](https://github.com/go-gitea/gitea/assets/15185051/24e0fa28-369b-4bb9-807a-ee823b46ba81)

delvh commented on templates/repo/issue/milestone_issues.tmpl:
```diff
@@ -39,7 +39,10 @@
 				&nbsp;
-				<b>{{.locale.Tr "repo.milestones.completeness" .Milestone.Completeness}}</b>
+				<div class="gt-df gt-ac">
+					<b class="gt-mr-3">{{.locale.Tr "repo.milestones.completeness" .Milestone.Completeness}}</b>
```
```suggestion
					<strong class="gt-mr-3">{{.locale.Tr "repo.milestones.completeness" .Milestone.Completeness}}</strong>
```

silverwind commented:
It should be on the same line, like in the issues list, and decolored:

<img width="540" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/854f9655-d523-4558-904f-2b236cba2855">

silverwind commented:
I can probably tweak this later. Keep note milestone and projects lists are generally in need of a rewrite to a template similar to issue list.

JakobDev commented:
> I can probably tweak this later. Keep note milestone and projects lists are generally in need of a rewrite to a template similar to issue list.

I just saw the the Progress bar, which is shown on the Milestone list is not shown on the Page of a single Milestne, so I added that. If you want to do more than that, I can close this PR so you can work n this.

> It should be on the same line, like in the issues list, and decolored:

This is shown in the Header, not in the List. The text was always in this Position. I just added the Bar next to it

silverwind commented:
Ah, I misread that screenshot. It indeed is on the single page, but I still can improve styling a bit there.

silverwind commented:
Tweaks done:

<img width="1269" alt="Screenshot 2023-06-02 at 20 32 04" src="https://github.com/go-gitea/gitea/assets/115237/cef839cd-1fe0-4947-b54b-609c48268634">
<img width="1267" alt="Screenshot 2023-06-02 at 20 32 21" src="https://github.com/go-gitea/gitea/assets/115237/63fb2a96-b0c5-4070-95c0-a699a7f0ddf2">

silverwind commented on options/locale/locale_en-US.ini:
```diff
@@ -1755,7 +1755,7 @@ milestones.no_due_date = No due date
 milestones.close = Close
 milestones.new_subheader = Milestones organize issues and track progress.
-milestones.completeness = %d%% Completed
+milestones.completeness = <strong>%d%%</strong> Completed
```
I would have really liked to use `{percentage} Completed` here where the template code takes care of percentage rendering, but it's currently not possible with this translation system to have such tokens.

silverwind replied:
I could put `%s completed` and avoid the HTML in the translation, but that might confuse translators because they don't know that `%s` is supposed to be a percentage.

JakobDev replied:
Is it not preferred, do don't change existing translations?

delvh replied:
Well… Yesn't.
The problem is simply that new strings (which is the case for modified strings) can be untranslated for quite some time.

wxiaoguang commented on web_src/css/repo.css:
```diff
@@ -1225,6 +1225,11 @@
 }
 
+.milestone-progress-big {
+  width: min(420px, 96vw);
```
Sorry but ... why "width: min" here?

* Does "420px" overflow some small width screens?
* Could the "96vw" be replaced by "flex: 1" ?

silverwind replied:
I don't know how small screens can get, but `min` confines the bar neatly to the screen with equal margins. Could also use q media query to achieve the same effect without `vw` which is a bit cleaner.
